go.mod

```diff
@@ -45,14 +45,14 @@ require (
 	golang.org/x/sync v0.8.0
 	gopkg.in/ini.v1 v1.67.0
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.30.3
-	k8s.io/apiextensions-apiserver v0.30.3
-	k8s.io/apimachinery v0.30.3
+	k8s.io/api v0.31.0
+	k8s.io/apiextensions-apiserver v0.31.0
+	k8s.io/apimachinery v0.31.0
 	k8s.io/cli-runtime v0.30.3
-	k8s.io/client-go v0.30.3
+	k8s.io/client-go v0.31.0
 	k8s.io/cloud-provider v0.30.3
-	k8s.io/utils v0.0.0-20240502163921-fe8a2dddb1d0
-	sigs.k8s.io/controller-runtime v0.18.4
+	k8s.io/utils v0.0.0-20240711033017-18e509b52bc8
+	sigs.k8s.io/controller-runtime v0.19.0
```

Comment:
I'm having trouble finding the mechanism by which updating `github.com/ceph/ceph-csi-operator/api` (line 21) is forcing us to update controller-runtime. 

CSI operator already has a module specific for API, and I don't see c-r in there. https://github.com/ceph/ceph-csi-operator/blob/main/api/go.mod

Is there something else forcing us to update this?

Reply:
I did go get csi-operator and go mod tidy...it brought all these dependencies.

Reply:
am not sure its brought because of the ceph-csi-operator and even it uses kube API 1.30 not 1.31  and also as @BlaineEXE mentioned it wont use controller-runtime. let me know if its happening because of csi-operator i will take a look at it.

Reply:
I'll check next time